src/riak_core_coverage_fsm.erl

```diff
@@ -332,27 +331,42 @@ terminate(Reason, _StateName, _State) ->
 code_change(_OldVsn, StateName, State, _Extra) ->
     {ok, StateName, State}.
 
-plan_callback(Mod, Exports) ->
-    case exports(plan, Exports) of
-        true ->
+%% This is to avoid expensive module_info calls, which were consuming
+%% 30% of the query-path time for small queries
+%%
+%% Instead we try Mod:plan/2, if undef, we define it.  On success or
+%% match error on atoms, we pass on Mod:plan/2 
+
+plan_callback(Mod) ->
+    try
+        Mod:plan(a, b),
+        fun(CoverageVNodes, ModState) ->
+                Mod:plan(CoverageVNodes, ModState) end
+    catch
+        error:undef ->
+            fun(_, ModState) ->
+                    {ok, ModState} end;
+        _:_ -> %% If Mod:plan(a, b) fails on atoms
```

Comment:
To avoid messing it up later I suggest we consider writing:

```Erlang
SuccessFun = fun(CoverageVNodes, ModState) ->
                            Mod:plan(CoverageVNodes, ModState) end
try
    SuccessFun(a, b),
    SuccessFun
catch
    ...
    _:_ -> 
        SuccesFun
```

Just to ensure that it is clear that the function is the same.

And perhaps we could write
```Erlang
SuccessFun = fun Mod:plan/2,
```